src/NuGet.Clients/PackageManagement.VisualStudio/IDE/NuGetAndEnvDTEProjectCache.cs

```diff
@@ -124,6 +125,9 @@ public EnvDTEProjectName AddProject(EnvDTEProject project, VSNuGetProjectFactory
         {
```

Comment:
That's not practically 100% possible, so I prefer not to go in this mode. It is something we can start doing in the future when we build a proper public API